### DISCUSSION TITLE
move video directory

### DISCUSSION THREAD

**Discussion by Seggew** [talmolab/sleap#1742] [Help!]
Hi there, when I first imported the videos I'm working with, they were in the wrong folder. When I move them to the right one I cannot open my model. How can I change the directory to these videos so that the model can access them from where I want to move them to?

**Comment by Marielamend** [comment]
Hi @Seggew!
You can replace the video path, by going to File > Replace Videos... Then it will open a window (example below) asking to double-click on a file and select the file you want to replace, then you can navigate and select the correct video under the right folder. You can do this for each video file, make sure you select the matching video for its respective file. 

<img width="366" alt="Screenshot 2024-04-11 at 12 02 10 PM" src="https://github.com/talmolab/sleap/assets/120222245/73d0c06c-b6e7-471f-bf56-66511e378d45">


If you have any other questions let us know!

Best, 
Mariela

**Reply by wegmor** [reply]
Is there a way to do this in a script using the python module (e.g. something like labels.replace_video)?

**Reply by talmo** [reply]
Hi @wegmor,

Yes, but I'd recommend doing this using our dedicated `sleap-io` package ([docs](https://io.sleap.ai)) which produces core SLEAP-compatible files but is easier to work with:

```py
import sleap_io as sio

labels = sio.load_slp("labels.v001.slp")
labels.replace_filenames(["new_video1.mp4", "new_video2.mp4"])
labels.save("labels.v002.slp")
```

There are some more advanced methods for replacing paths too if you check out the [`Labels.replace_filenames()`](https://io.sleap.ai/latest/model/#sleap_io.Labels.replace_filenames) docs.

Let us know if you need any help!

Cheers,

Talmo
